proposals/lip-0050.md

```diff
@@ -171,14 +171,36 @@ registerBLSKeyParamsSchema = {
         "proofOfPossession": {
             "dataType": "bytes",
             "fieldNumber": 2
+        },
+        "generatorKey": {
+            "dataType": "bytes",
+            "fieldNumber": 3
         }
     }
 }
 ```
 
 ##### Execution
 
-Executing a transaction `trs` triggering an register BLS key command is done by calling the function `setValidatorBLSKey(validatorAddress, trs.params.proofOfPossession, trs.params.blsKey)` where `validatorAddress` is the 20-byte address derived from `trs.senderPublicKey` and `setValidatorBLSKey` is defined in the [Validators module][lip-0044#setValidatoBLSKey]. If this function returns false, the transaction is invalid.
+Executing a transaction `trs` triggering an register keys command is done by calling the logic below.
+
+```python
+validatorAddress = 20-byte address derived from trs.senderPublicKey
+
+if validator.getValidatorAccount(validatorAddress) does not exists :
+    trs fails
+    
+if validator.getValidatorAccount(validatorAddress).generatorKey == INVALID_ED25519_KEY:
+    setGeneratorKey = validator.setValidatorGeneratorKey(validatorAddress, trs.params.generatorKey)
+elif validator.getValidatorAccount(validatorAddress).generatorKey == trs.params.generatorKey):
+    setGeneratorKey = True
+    
+setBLSKey = validator.setValidatorBLSKey(validatorAddress, trs.params.proofOfPossession, trs.params.blsKey)
+
+return (setBLSKey and setGeneratorKey)
+```
+
+Those functions are defined in the [Validators module][lip-0044#setValidatoBLSKey]. If the above logic returns false, the transaction is invalid.
```

Comment:
> If the above logic returns false, the transaction is invalid.

I think, in the spirit of error handling we should raise an error if the transaction is invalid. 
For example, consider the following piece of code:

```python
existingKey =  validator.getValidatorAccount(validatorAddress).generatorKey
if existingKey is empty or existingKey == INVALID_ED25519_KEY:
     setGeneratorKey = validator.setValidatorGeneratorKey(validatorAddress, trs.params.generatorKey)
     if not setGeneratorKey:
         raise Error("Error when setting generator key")
else:
    raise Error("Generator key is already set")
```

Or maybe this logic could be finalized after we introduce the error handling logic to Validators module.